sdcm/utils/features.py

```diff
@@ -0,0 +1,78 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+import logging
+
+from cassandra.cluster import Session  # pylint: disable=no-name-in-module
+
+CONSISTENT_TOPOLOGY_CHANGES_FEATURE = "SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES"
+CONSISTENT_CLUSTER_MANAGEMENT_FEATURE = "SUPPORTS_RAFT_CLUSTER_MANAGEMENT"
+TABLETS_FEATURE = "TABLETS"
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_supported_features(session: Session) -> list[str]:
+    """
+    helper function to get supported_features from a running cluster,
+    if you need from a specific node use `patient_exclusive_cql_connection` session
+    """
+    result = session.execute("SELECT supported_features FROM system.local WHERE key='local'").one()
+    # NOTE: since row_factory can be different on different tests, we need to support multiple options
+    if isinstance(result, dict):
+        result = result["supported_features"].split(",")
+    elif isinstance(result, tuple):
+        result = result[0].split(",")
+    else:
+        raise NotImplementedError(f"unsupported row_factory={session.row_factory}")
+    LOGGER.debug("Supported features %s", result)
+    return result
+
+
+def get_enabled_features(session: Session) -> list[str]:
+    """
+    helper function to get supported_features from a running cluster,
+    if you need from a specific node use `patient_exclusive_cql_connection` session
+    """
+    result = session.execute("SELECT value FROM system.scylla_local WHERE key='enabled_features'").one()
+    # NOTE: since row_factory can be different on different tests, we need to support multiple options
+    if isinstance(result, dict):
+        result = result["value"].split(",")
+    elif isinstance(result, tuple):
+        result = result[0].split(",")
+    else:
+        raise NotImplementedError(f"unsupported row_factory={session.row_factory}")
+    LOGGER.debug("Enabled features %s", result)
+    return result
+
+
+def is_raft_schema_changes_feature_enabled(session: Session) -> bool:
+    """ Check whether raft topology feature enabled
+    if you need from a specific node use `patient_exclusive_cql_connection` session
+    """
+
+    return CONSISTENT_CLUSTER_MANAGEMENT_FEATURE in get_enabled_features(session)
+
+
+def is_raft_topology_changes_feature_enabled(session: Session) -> bool:
+    """ Check whether raft topology feature enabled
+    if you need from a specific node use `patient_exclusive_cql_connection` session
+    """
+
+    return CONSISTENT_TOPOLOGY_CHANGES_FEATURE in get_enabled_features(session)
+
+
+def is_tablets_feature_enabled(session: Session) -> bool:
+    """ Check whether tablets enabled
+    if you need from a specific node use `patient_exclusive_cql_connection` session
+    """
+    return TABLETS_FEATURE in get_enabled_features(session)
```

Comment:
is it that simple? Shouldn't we verify all the nodes in that case?

Reply:
we can do it on cluster level, by adding appropriate method with exclusive session. But this could be done (and will be in next pr)